src/shogun/io/SGIO.h

```diff
@@ -231,34 +195,38 @@ class SGIO
 	public:
 		/** default constructor */
 		SGIO();
-		/** copy constructor */
-		SGIO(const SGIO& orig);
 
 		/** destructor */
 		virtual ~SGIO();
 
+		/** (re)initializes the default (asynchronous) logger
+		 * 
+		 * @param queue_size size of the message queue
+		 * @param n_threads number of logging threads
+		 */
+		void init_default_logger(uint64_t queue_size=128, uint64_t n_threads=1);
```

Comment:
why the n threads parameter? wouldnt we use the shogun number of threads?

Reply:
to point out that these are different threads

Reply:
could you give me some use cases for those threads?

Reply:
In general, if the logging code is slow, the performance will take a great hit. And in multithreaded code, it will get even slower because of the mutex locks. Async loggers solve this by having multiple logging queues in different threads so that logging calls are nonblocking.